third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -373,7 +373,15 @@ impl<'env> Generator<'env> {
                 }
                 // If there is a binding, assign the pattern
                 if let Some(binding) = opt_binding {
-                    self.gen_assign(pat.node_id(), pat, binding, Some(&scope));
+                    if let Pattern::Var(var_id, sym) = pat {
```

Comment:
If you did this specialization inside `gen_assign` then you'd also get this benefit in the the multi-var `(a, b, c) = (exp1, exp2, exp3)` case (since `gen_assign` is invoked inside `gen_tuple_assign` when the exps don't have overlapping variables.